lib/filewatch/sincedb_collection.rb

```diff
@@ -225,8 +225,13 @@ def sincedb_write(time = Time.now)
 
     # @return expired keys
     def atomic_write(time)
-      FileHelper.write_atomically(@full_path) do |io|
-        @serializer.serialize(@sincedb, io, time.to_f)
+      begin
+        FileHelper.write_atomically(@full_path) do |io|
+          @serializer.serialize(@sincedb, io, time.to_f)
+        end
+      rescue => e
+        logger.warn("sincedb_write: unable to write atomically, falling back to non-atomic write: #{path} error:", :exception => e.class, :message => e.message)
```

Comment:
do you foresee atomic write potentially recovering, later?
if not maybe we would want the switch permanently to non-atomic ... (on the caller end).
guess this is rare, still seems like we would produce a lot of the same warn-ings (which might be fine :thinking:).

Reply:
It's a good question. I think it might make sense to make the change 'sticky' in the event of permission errors, which are probably unlikely to recover later, and leave as is in the event of other errors.